File: files/samples/Video_Stream.py
from native import app, ui

# Translated Python Code

# Called when application is started.
def OnStart():
    global player, skb
    # Create layout that fills the screen.
    lay = ui.createLayout( "Linear", "FillXY,VCenter" )

    # Create video view.
    player = app.createVideoView( 0.9, 0.4 )
    lay.addChild( player )
    player.setOnReady( player_OnReady )
    player.setOnComplete( player_OnComplete )
    player.setOnError( player_OnError )

    # Create video list.
    spin = ui.createSpinner( "...,NASA,BUNNY" )
    spin.setSize( 0.8, -1 )
    spin.setOnTouch( spn_OnTouch )
    lay.addChild( spin )

    # Create Play button.
    btnPlay = ui.createButton( "Play", 0.4, 0.1 )
    btnPlay.setMargins( 0, 0.05, 0, 0 )
    btnPlay.setOnTouch( btnPlay_OnTouch )
    lay.addChild( btnPlay )

    # Create Pause button.
    btnPause = ui.createButton( "Pause", 0.4, 0.1 )
    btnPause.setOnTouch( btnPause_OnTouch )
    lay.addChild( btnPause )

    # Create seek bar and add to layout.
    skb = ui.createSeekBar( 0.8, -1 )
    skb.setMargins( 0, 0.05, 0, 0 )
    skb.setRange( 1.0 )
    skb.setOnTouch( skb_OnTouch )
    lay.addChild( skb )

    # Add main layout to app.
    app.addLayout( lay )

    # Start timer to update seek bar every second.
    setInterval( "Update()", 1000 )

# Handle file select.
def spn_OnTouch(item):
    if item == "NASA":
        app.showProgress( "Buffering..." )
        setTimeout( "app.hideProgress()", 7000 )
        player.setFile( "https://archive.org/download/NASAMissions/NASA Missions.mp4" )
    elif item == "BUNNY":
        app.showProgress( "Buffering..." )
        setTimeout( "app.hideProgress()", 10000 )
        player.setFile( "rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mov" )
        skb.setVisibility( "Show" )

# Called when file is ready to play.
def player_OnReady():
    app.hideProgress()
    player.play()

# Called when playback has finished.
def player_OnComplete():
    app.showPopup( "Finished" )

# Called when an error occurs.
def player_OnError():
    app.showPopup( "Error" )

# Handle Play button.
def btnPlay_OnTouch():
    player.play()

# Handle Pause button.
def btnPause_OnTouch():
    player.pause()

# Called when user touches the seek bar.
def skb_OnTouch(value):
    player.seekTo( dur * value )

# Update seek bar.
def Update():
    dur = player.getDuration()
    prog = player.getPosition()
    if dur:
        skb.setValue( prog / dur )

File: files/samples/js/Video_Stream.js

//Called when application is started.
function OnStart()
{   
    //Create layout that fills the screen.
    lay = app.CreateLayout( "Linear", "FillXY,VCenter" )
        
    //Create video view.
    player = app.CreateVideoView( 0.9, 0.4 )
    lay.AddChild( player )
    player.SetOnReady( player_OnReady )
    player.SetOnComplete( player_OnComplete )
	player.SetOnError( player_OnError )
    
    //Create video list.
    spin = app.CreateSpinner( "...,NASA,BUNNY" )
    spin.SetSize( 0.8, -1 )
    spin.SetOnTouch( spn_OnTouch )
    lay.AddChild( spin )
    
    //Create Play button.
    btnPlay = app.CreateButton( "Play", 0.4, 0.1 )
    btnPlay.SetMargins( 0, 0.05, 0, 0 )
    btnPlay.SetOnTouch( btnPlay_OnTouch )
    lay.AddChild( btnPlay ) 
    
    //Create Pause button.
    btnPause = app.CreateButton( "Pause", 0.4, 0.1 )
    btnPause.SetOnTouch( btnPause_OnTouch )
    lay.AddChild( btnPause ) 
	
	//Create seek bar and add to layout.
	skb = app.CreateSeekBar( 0.8, -1 )
	skb.SetMargins( 0, 0.05, 0, 0 )
	skb.SetRange( 1.0 )
	skb.SetOnTouch( skb_OnTouch )
	lay.AddChild( skb )
    
    //Add main layout to app.
    app.AddLayout( lay )
	
	//Start timer to update seek bar every second.
	setInterval( "Update()", 1000 )
}

//Handle file select.
function spn_OnTouch( item )
{
    if( item=="NASA" ) {
		app.ShowProgress( "Buffering..." )
		setTimeout( "app.HideProgress()", 7000 )
		player.SetFile( "https://archive.org/download/NASAMissions/NASA Missions.mp4" )
	}
	else if( item=="BUNNY" ) {
		app.ShowProgress( "Buffering..." )
		setTimeout( "app.HideProgress()", 10000 )
		player.SetFile( "rtsp://wowzaec2demo.streamlock.net/vod/mp4:BigBuckBunny_115k.mov")
		skb.SetVisibility( "Show" )
	}
}

//Called when file is ready to play.
function player_OnReady()
{
	app.HideProgress()
    player.Play()
}

//Called when playback has finished.
function player_OnComplete()
{
    app.ShowPopup( "Finished" )
}

//Called when an error occurs.
function player_OnError()
{
    app.ShowPopup( "Error" )
}

//Handle Play button.
function btnPlay_OnTouch() 
{
    player.Play()
}

//Handle Pause button.
function btnPause_OnTouch() 
{
    player.Pause()
}

//Called when user touches the seek bar.
function skb_OnTouch( value )
{
	player.SeekTo( dur * value )
}

//Update seek bar.
function Update()
{
	dur = player.GetDuration()
	prog = player.GetPosition()
	if( dur ) skb.SetValue( prog / dur )
}

